lib/Frontend/Frontend.cpp

```diff
@@ -311,7 +311,13 @@ void CompilerInstance::performSema() {
   Module *importedHeaderModule = nullptr;
   StringRef implicitHeaderPath = options.ImplicitObjCHeaderPath;
   if (!implicitHeaderPath.empty()) {
-    if (!clangImporter->importBridgingHeader(implicitHeaderPath, MainModule)) {
+    bool failed = false;
+    if (llvm::sys::path::extension(implicitHeaderPath).endswith(PCH_EXTENSION)) {
```

Comment:
I think I would sink this logic inside `importBridgingHeader`. The less the frontend knows about gory Clang details the better.